### ISSUE TITLE
`type-annotation-spacing`: 'Value "ignore" should be object' error

### ISSUE THREAD

**Issue by m-jung** [eslint-stylistic/eslint-stylistic#1240]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Using [`overrides: { property: 'ignore' }`](https://eslint.style/rules/type-annotation-spacing#property) i get the error _"Value "ignore" should be object."_

The documentation for [`property`](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/packages/eslint-plugin/rules/type-annotation-spacing/README.md#property)says:

> Examples of correct code for this rule with the `{ "before": false, "after": false, "overrides": { "property": "ignore" } }` option. Where no property type spacing is changed so it won't conflict with key-spacing or any spacing rule.

### Config
```javascript
import { defineConfig } from "eslint/config";
import stylistic from '@stylistic/eslint-plugin';

export default defineConfig([
  {
    files  : ["**/*.js"],
    plugins: { '@stylistic': stylistic },
    rules  : {
      '@stylistic/type-annotation-spacing': [
        'error',
        {
          before   : false,
          after    : true, 
          overrides: { property: 'ignore' },
        },
      ],
    },
  }
]);
```

### Expected behavior
I'd expect running `npx eslint .`  on the following JS not throwing any errors.

```javascript
export default {
    lorem       : 'ipsum',
    dolorSitAmet: 'consetetur sadipscing elitr',
}
```

### Actual behavior

```text
Error: Key "rules": Key "@stylistic/type-annotation-spacing":
        Value "ignore" should be object.
```

### Reproduction

https://stackblitz.com/edit/github-dehiwzsd-ugypkebd

### System Info

```shell
System:
    OS: Windows 11 10.0.26100
  Binaries:
    Node: 26.5.0
    npm: 11.17.0
  npmPackages:
    @stylistic/eslint-plugin: 5.10.0 => 5.10.0
    eslint: 10.7.0 => 10.7.0
```

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by 9romise** [comment]
Sorry for the confusion. This is already in the v6.0.0 beta. Feel free to try it out by upgrading to 6.0 beta. We’re planning to release the stable 6.0 after the next beta, as long as nothing serious pops up.
https://eslint.style/rules/type-annotation-spacing#changelog
